### ISSUE TITLE
Bug: inconsistency between Tensorflow, CNTK and Theano backends

### ISSUE THREAD

**Issue by Dandy-John** [keras-team/keras#13572]
## Environment settings

Nvidia Driver Version: 430.50

CUDA Version: 10.1

GeForce RTX 2080 Ti

Ubuntu 16.04.6 LTS

| Library                        | Version  |       |
| ------------------------------ | -------- | ----- |
|libgpuarray               |0.7.6                |h14c3975_0|
|pygpu                     |0.7.6            |py36h035aef0_0|
|tensorflow-gpu            |1.14.0                    |\<pip>|
|tensorflow-estimator      |1.14.0                    |\<pip>|
|Theano                    |1.0.4                     |\<pip>|
|cntk-gpu                  |2.7                       |\<pip>|
|numpy                     |1.14.6           |py36h3b04361_4|
|numpy-base                |1.14.6           |py36h81de0dd_4|
|keras-applications        |1.0.8                      |py_0|
|keras-preprocessing       |1.1.0                      |py_1|

---

### Description

lstm-sinewave_origin0-MLA2.h5 is a model that can be used to predict sine wave. I loaded the same model on different backends(tensorflow, CNTK and theano), and the prediction accuracy of CNTK are different from that of tensorflow and theano under same input. We calcuate the MSE between the output of the model and the ground truth. MSE can be calculated as:
![image](https://user-images.githubusercontent.com/17698785/69396888-c98f1980-0d1e-11ea-9087-fbc7aecdd01d.png)

The MSE value of tensorflow and theano are 0.4134036719799042 while the value of CNTK is 0.2730630080302556.

I put input data into the model and recorded the ouputs of each layers to see how this happens. It seems that the outputs on tensorflow and theano are very close, but for CNTK,
the outputs of the first layer and all the layers after this layer are very different from tensorflow and theano. 

The results are attached below.

![image](https://user-images.githubusercontent.com/17698785/69396810-82088d80-0d1e-11ea-8dc8-d5edd46d26f0.png)

The delta calumn shows the discrepancy, which is calculated as:
![image](https://user-images.githubusercontent.com/17698785/69396896-d3b11800-0d1e-11ea-8a0a-69f03e98a0ac.png)

We attach the .h5 file, input and the code below.

1. to see the MSE value:
   ```
   python get_model_accurary.py [backend name]
   ```

2. to see the output of intermediate layers:
   ```
   python get_layers_output.py [backend name]
   ```
3. bug localization:
   ```
   python get_layers_deviation.py
   ```
[20191114-lstm.zip](https://github.com/keras-team/keras/files/3877931/20191114-lstm.zip)
